sdk/keyvault/Azure.Security.KeyVault.Keys/tests/KeysTestBase.cs

```diff
@@ -0,0 +1,200 @@
+                }
+            }
+            finally
+            {
+                _keysToCleanup.Clear();
+            }
+        }
+
+        protected void RegisterForCleanup(KeyBase key, bool delete = true)
+        {
+            _keysToCleanup.Enqueue((key, delete));
+        }
+
+        protected void AssertKeysEqual(Key exp, Key act)
+        {
+            AssertKeyMaterialEqual(exp.KeyMaterial, act.KeyMaterial);
+            AssertKeysEqual((KeyBase)exp, (KeyBase)act);
+        }
+
+        private void AssertKeyMaterialEqual(JsonWebKey exp, JsonWebKey act)
+        {
+            Assert.AreEqual(exp.KeyId, act.KeyId);
+            Assert.AreEqual(exp.KeyType, act.KeyType);
+            Assert.IsTrue(AreEqual(exp.KeyOps, act.KeyOps));
+            Assert.AreEqual(exp.CurveName, act.CurveName);
+            Assert.AreEqual(exp.K, act.K);
+            Assert.AreEqual(exp.N, act.N);
+            Assert.AreEqual(exp.E, act.E);
+            Assert.AreEqual(exp.X, act.X);
+            Assert.AreEqual(exp.Y, act.Y);
+            Assert.AreEqual(exp.D, act.D);
+            Assert.AreEqual(exp.DP, act.DP);
+            Assert.AreEqual(exp.DQ, act.DQ);
+            Assert.AreEqual(exp.QI, act.QI);
+            Assert.AreEqual(exp.P, act.P);
+            Assert.AreEqual(exp.Q, act.Q);
+            Assert.AreEqual(exp.T, act.T);
+        }
+
+        protected void AssertKeysEqual(KeyBase exp, KeyBase act)
+        {
+            Assert.AreEqual(exp.Managed, act.Managed);
+            Assert.AreEqual(exp.RecoveryLevel, act.RecoveryLevel);
+            Assert.AreEqual(exp.Expires, act.Expires);
+            Assert.AreEqual(exp.NotBefore, act.NotBefore);
+            Assert.IsTrue(AreEqual(exp.Tags, act.Tags));
+        }
+
+        private static bool AreEqual(IList<KeyOperations> exp, IList<KeyOperations> act)
+        {
+            if (exp == null && act == null)
+                return true;
+
+            if (exp.Count != act.Count)
+                return false;
+
+            for (var i = 0; i < exp.Count; ++i)
+                if (exp[i] != act[i])
+                    return false;
+
+            return true;
+        }
+
+        private static bool AreEqual(IDictionary<string, string> exp, IDictionary<string, string> act)
+        {
+            if (exp == null && act == null)
+                return true;
+
+            if (exp?.Count != act?.Count)
+                return false;
+
+            foreach (var pair in exp)
+            {
+                if (!act.TryGetValue(pair.Key, out string value)) return false;
+                if (!string.Equals(value, pair.Value)) return false;
+            }
+            return true;
+        }
+
+        protected Task WaitForDeletedKey(string name)
+        {
+            if (Mode == RecordedTestMode.Playback)
+            {
+                return Task.CompletedTask;
+            }
+
+            using (Recording.DisableRecording())
```

Comment:
Any specific reason to not record the polling attempts?  I think it's fine to keep this, but want to make sure I'm not doing something dumb in  the Storage tests.

Reply:
Not that I know of. Didn't see much value but @pakrym might know better as he used this pattern when implementing Secrets tests